File: sample/src/main/java/com/infullmobile/android/infullmvp/sample/activity/SampleActivityPresenter.java
package com.infullmobile.android.infullmvp.sample.activity;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.infullmobile.android.infullmvp.Presenter;

public class SampleActivityPresenter extends Presenter<SampleActivityView> {

    private final SampleActivityModel sampleActivityModel;

    public SampleActivityPresenter(@NonNull SampleActivityView presentedView,
                                   @NonNull SampleActivityModel sampleActivityModel) {
        super(presentedView);
        this.sampleActivityModel = sampleActivityModel;
    }

    @Override
    protected void bind(@NonNull Bundle bundle, @NonNull Bundle savedInstanceState, @Nullable Uri intentData) {
        getPresentedView().setMessage(sampleActivityModel.getMessage());
    }
}

File: sample-kotlin/src/main/java/com/infullmobile/android/infullmvp/sample/activity/SampleActivityPresenter.kt
package com.infullmobile.android.infullmvp.sample.activity

import android.net.Uri
import android.os.Bundle
import com.infullmobile.android.infullmvp.Presenter
import com.infullmobile.android.infullmvp.sample.Navigation

class SampleActivityPresenter(
        presentedView: SampleActivityView,
        private val sampleActivityModel: SampleActivityModel,
        val navigation: Navigation
)
    : Presenter<SampleActivityView>(presentedView) {

    override fun bind(intentBundle: Bundle, savedInstanceState: Bundle, intentData: Uri?) {
        presentedView.message = sampleActivityModel.message
    }

    fun navigateToCustomView() {
        navigation.customViewScreen().start()
    }
}